src/data/finset/powerset.lean

```diff
@@ -117,5 +117,37 @@ begin
     exact mem_powerset.mpr (mem_powerset_len.mp ha).1, }
 end
 
+lemma powerset_len_sup [decidable_eq α] (u : finset α) (n : ℕ) (hn : n < u.card) :
```

Comment:
I couldn't find the best place for this. To put it in this file, I had to change the import to import `data.finset.lattice`.

This proof can probably be generalized to other "monadic" like functions, but that seemed to be too yak-shavy here.

Reply:
I can't think of any other monadic-like functions which generalize this naturally. Did you have some in mind? (Maybe an `inf` version?) 

Otherwise, I think by breaking this up we could keep the powerset specific parts of the argument in `powerset.lean` and move the rest into `finset.lattice.lean`.